packages/@uppy/status-bar/src/StatusBar.jsx

```diff
@@ -8,25 +8,56 @@ import StatusBarUI from './StatusBarUI.jsx'
 import packageJson from '../package.json'
 import locale from './locale.js'
 
-function getTotalSpeed (files) {
-  let totalSpeed = 0
-  files.forEach((file) => {
-    totalSpeed += getSpeed(file.progress)
-  })
-  return totalSpeed
+const speedFilterHalfLife = 2
+const ETAFilterHalfLife = 2
+
+let previousTotalBytes = 0
+function getCurrentSpeed (files, dt) {
+  const totalBytes = files.reduce((total, file) => {
+    return total + file.progress.bytesUploaded
+  }, 0)
+  if (totalBytes === 0) {
+    return 0
+  }
+  const newlyUploadedBytes = totalBytes - previousTotalBytes
+  previousTotalBytes = totalBytes
+  return newlyUploadedBytes / dt
 }
 
-function getTotalETA (files) {
-  const totalSpeed = getTotalSpeed(files)
-  if (totalSpeed === 0) {
+let previousSpeed = null
```

Comment:
We probably need to use generators to avoid "leaking" variables here and there.

Reply:
I tried that in https://github.com/transloadit/uppy/pull/4458/commits/1a0ea4970e53ef589a1abdea1ac848f31dc120a3, and I simplified it further in https://github.com/transloadit/uppy/pull/4458/commits/d1445e038de8e258f7b521b5a661cb688f551260, PTAL